### ISSUE TITLE
version 0.28.24 broken

### ISSUE THREAD

**Issue by garretwilson** [Glavin001/atom-beautify#839]
OK, what just happened here? My Atom upgraded to atom-beautify 0.28.24, and suddenly my HTML files are formatted crazy!!!! Everything looks horrible; even stuff that was already formatted:

````html
<ul>
  <li><dfn>annotated tag</dfn></li>
  <li><dfn>detached <code>HEAD</code></dfn></li>
</ul>
````

...now looks like this!

````html
<ul>
  <li>
    <dfn>annotated tag</dfn>
  </li>
  <li>
    <dfn>detached
      <code>HEAD</code>
    </dfn>
  </li>
</ul>
````

I have a `.jsbeautifyrc` file, so no custom settings should have changed. But it seems to be completely ignoring this settings file, now.

Looking in the config, I don't even see a **Language Config - HTML** section anymore.

What just happened? Somebody majorly broke something. It could have been me---but I'm pretty sure all I did was upgrade to the latest atom-beautify version...

**Comment by prettydiff** [comment]
Would you mind going through the template: https://github.com/Glavin001/atom-beautify/blob/master/ISSUE_TEMPLATE.md

The debug log would be really helpful so that we can know which beautifier you are using, options are selected, and steps to reproduce.

**Comment by garretwilson** [comment]
Maybe we could start out with something simple first. Could you or @Glavin001 kindly tell me:

* Have there been major changes in 0.28.24?
* Should I expect to see the **Language Config - HTML** section, or has it been removed in 0.28.24?
* Should atom-beautify still honor my `.jsbeautifyrc` file?

(Yes, I did an uninstall and then a reinstall.)

**Comment by garretwilson** [comment]
So I'm trying to downgrade to a lower version. I notice that https://github.com/Glavin001/atom-beautify/releases indicates there was a 0.28.22 version, but `apm` doesn't seem to find it; maybe someone removed it.

**Comment by garretwilson** [comment]
I downgraded to version 0.28.22, and it "works" again (and by "works" I mean it mostly formats HTML, using all the workarounds for all the other bugs I've reported here and on js-beautify.

So in my opinion you have a huge, breaking bug in 0.28.4. No, @prettydiff , I'm sorry but this time I'm not going through all the template stuff; this one you and @Glavin001 should be able to track down without the template, although I'll be happy to answer any questions you may have. Good luck.

**Comment by Glavin001** [comment]
> I downgraded to version 0.28.22, and it "works" again

Excellent. Thank you for confirming it is after v0.28.22.

Did you restart or reload Atom between installation/updates? I used to find that `apm` would cache sometimes and updating would break, and a clear restart/reload of Atom would resolve all of the issues for users. Therefore I now always recommend at least reloading Atom between installs/updates.

> apm doesn't seem to find [version 0.28.23]

v0.28.23 was unpublished from `apm`.

> Have there been major changes in 0.28.24?

I reverted a commit in v0.28.23 that was mistakenly merged and pushed: https://github.com/Glavin001/atom-beautify/compare/v0.28.23...v0.28.24

Many small changes occurred in v0.28.23:
https://github.com/Glavin001/atom-beautify/compare/v0.28.22...v0.28.23
Which would also be relevant to v0.28.24 with exception of the commit I reverted: https://github.com/Glavin001/atom-beautify/commit/d97b2d2e4db3e52e62b5509b7bacbbcc3bea0e56

> Should I expect to see the Language Config - HTML section, or has it been removed in 0.28.24?

Yup, it definitely should be there. Applicable code is still in v0.28.24: https://github.com/Glavin001/atom-beautify/blob/v0.28.24/src/beautifiers/index.coffee#L256-L288

Can anyone (/cc @prettydiff ) else confirm that `Language Config` or other options are missing in their Atom Beautify package settings in v0.28.24? I see everything as expected and cannot reproduce this.

> Should atom-beautify still honor my .jsbeautifyrc file?

Yes, it definitely still should. None of that core code has been touched.

> So in my opinion you have a huge, breaking bug in 0.28.4.

Atom Beautify definitely did in v0.28.23 -- my mistake, sorry about that everyone -- however it was reverted with v0.28.24.

>  I'm sorry but this time I'm not going through all the template stuff; this one you and @Glavin001 should be able to track down without the template, although I'll be happy to answer any questions you may have.

You have pretty much answered most of what the Issue template asks (Description, Expected Results, sample input, etc), except not as organized with headings. I can live with that. You've provided great information so far, and I thank you for that. The Issue Template is not so much targeting users like yourself so much as users who actually create issues with the title "Bug" and no description, etc... :stuck_out_tongue_winking_eye: 

However, I do draw the line at the `debug.md` for the editor debugging information, especially for issues like this.
It takes a few seconds to create a `debug.md` file that will answer most, if not all, questions I will be asking to help diagnose this or similar issues. Help me help you, and please provide your debugging information by following the instructions found at https://github.com/Glavin001/atom-beautify/blob/master/ISSUE_TEMPLATE.md#how-to-create-debugmd-gist
Using your sample input above is sufficient. 
Thank you.

---

### Notes

What I will be looking at in the `debug.md` is how the options are being picked up and flowing through Atom Beautify. It sounds like the options are not being carried through to the beautifier, or not even being detected from either package settings or `.jsbeautifyrc` file.

Travis CI would be passing for v0.28.24 if not for 2 failing tests for LESS and Riot.js that are unrelated issues (Pretty Diff update causing LESS code to be different).
While the tests currently do not test the package settings, they do test the `.jsbeautifyrc`. So since @garretwilson is using a `.jsbeautifyrc` then even the bug in v0.28.23 -- which changed the package settings -- should not influence and cause this unexpected behaviour.

**Comment by ttsirkia** [comment]
v.0.28.24 "Language config - HTML" is not visible, only the HTML formatting options.

**Comment by ttsirkia** [comment]
Actually there is a strange "Language config - XTemplate" between Haskell and Java. Should that be HTML?

**Comment by ttsirkia** [comment]
debug.md: https://gist.github.com/ttsirkia/3279f596987c08fa2d83

**Comment by Glavin001** [comment]
Thank you @ttsirkia.
Comparing the expected beautified code against your expected code I would suggest the option you need is `unformatted`.
Looking at your [`Final Options`](https://gist.github.com/ttsirkia/3279f596987c08fa2d83#final-options) I see that it is missing. However, looking at your `Pre-Transformed Options` I do see the `unformatted` option. This tells me that the beautifier is transforming the options and likely does not support the `unformatted` option.
I see that the beautifier selected for beautification is `Pretty Diff`: https://gist.github.com/ttsirkia/3279f596987c08fa2d83#original-file-to-be-beautified
Looking at the `Pretty Diff` beautifier code I can confirm that `unformatted` option is not supported: https://github.com/Glavin001/atom-beautify/blob/master/src/beautifiers/prettydiff.coffee#L8-L36

My recommendation is to switch to using `JS-Beautify` for HTML.
However, in saying that I notice in that `Language Config - HTML` is also missing from the documentation found at https://github.com/Glavin001/atom-beautify/blob/master/docs/options.md

So this leads me to check again how the `Language Config - ` options are built: https://github.com/Glavin001/atom-beautify/blob/master/src/beautifiers/index.coffee#L256-L285
It is important to note that there is not currently a check to see if an existing option is being overridden.

> Actually there is a strange "Language config - XTemplate" between Haskell and Java. Should that be HTML?

This is the line that makes it all clear!

I investigate `XTemplate` language and discover that it uses the same namespace as HTML!!!
See https://github.com/Glavin001/atom-beautify/blob/master/src/languages/xtemplate.coffee#L4
I should have reviewed Pull Request https://github.com/Glavin001/atom-beautify/pull/758 more thoroughly and I apologize for this.

I will be:
- [ ] fixing XTemplate language
- [ ] Adding a check to throw error message to user that there is a conflicting option that overrides an existing option

Great work everyone. Thanks for your input. I'll be publishing fixes shortly.

**Comment by ttsirkia** [comment]
The setting I used are more or less the default options. The current conflict also prevents changing the engine. Nice to hear that the information was helpful!

**Comment by Glavin001** [comment]
Published fix to v0.28.25. Thank you for your feedback and patience.
